app/models/user_preference.rb

```diff
@@ -0,0 +1,46 @@
+class UserPreference < ApplicationRecord
+  belongs_to :user
+  validate :user_preference_checks
+
+  # Whether the user receives emails whenever a bulk download succeeds.
+  RECEIVES_BULK_DOWNLOAD_SUCCESS_EMAILS = "receives_bulk_download_success_emails".freeze
+
+  # value must be "true" or "false".
+  DATA_TYPE_BOOLEAN = "boolean".freeze
+
+  # Stores metadata for various user preferences.
+  USER_PREFERENCE_METADATA = {
+    RECEIVES_BULK_DOWNLOAD_SUCCESS_EMAILS => {
+      default: "true",
+      description: "Receive an email whenever a bulk download succeeds",
+      allowed_feature: "bulk_downloads",
```

Comment:
maybe `required_allowed_feature` or `prerequisite_feature`? It read a little confusingly to me here

Reply:
Agree with jonathan. 

Separately, I prefer less redundant name `UserPreference::METADATA`.